sdk/trace/batch_span_processor_test.go

```diff
@@ -136,18 +131,32 @@ func TestNewBatchSpanProcessorWithOptions(t *testing.T) {
 			genNumSpans:    205,
 			waitTime:       waitTime,
 		},
+		{
+			name: "parallel span generation",
+			o: []sdktrace.BatchSpanProcessorOption{
+				sdktrace.WithScheduleDelayMillis(schDelay),
+				sdktrace.WithMaxQueueSize(200),
+			},
+			wantNumSpans:   200,
+			wantBatchCount: 1,
+			genNumSpans:    205,
+			waitTime:       waitTime,
+			parallel:       true,
+		},
 	}
 	for _, option := range options {
 		te := testBatchExporter{}
 		tp := basicProvider(t)
 		ssp := createAndRegisterBatchSP(t, option, &te)
 		if ssp == nil {
-			t.Errorf("%s: Error creating new instance of BatchSpanProcessor\n", option.name)
+			t.Fatalf("%s: Error creating new instance of BatchSpanProcessor\n", option.name)
 		}
 		tp.RegisterSpanProcessor(ssp)
 		tr := tp.Tracer("BatchSpanProcessorWithOptions")
 
-		generateSpan(t, tr, option)
+		wg := &sync.WaitGroup{}
```

Comment:
Is there a reason the WaitGroup isn't created and handled in `generateSpan` itself?

Reply:
there is no reason. Moved it to generateSpan.